sandbox/controller.go

```diff
@@ -0,0 +1,60 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package sandbox
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+)
+
+// Controller is an interface to manage a runtime sandbox instance (runtimes's SandboxManager)
+// SandboxRuntime is responsible for the sandbox instances lifecycle management.
+// When running the traditional containerd shim, the workflow looks as follows:
+// For each new task we're about to run:
+// 	1. Invoke `shim_binary --start` to obtain `TaskService` address (printed in stdout)
+// 	2. Call TaskService.RunContainer(id=1)
+// 	3. Exec `shim_binary --delete` to stop shim
+// 	4. Exec `shim_binary --start` again to obtain another `TaskService` address
+// 	5. TaskService.RunContainer(id=2)
+// 	6. Exec `shim_binary --delete` to stop shim
+//
+// When running in sandbox mode, shim must implement `SandboxService`. In sandbox mode shim lifetimes are managed manually.
+// 	1. Client calls `client.Controller.Start()` to launch new shim and create sandbox process
+// 	2. Run containers with `shim.TaskService.RunContainer(id=1)` and another one `shim.TaskService.RunContainer(id=2)`
+//	3. ... usual container lifecycle calls to `shim.TaskService`
+// 	4. Client calls shim to stop the VM with `client.SandboxService.Shutdown()`
```

Comment:
stop the sandbox?

Reply:
@mxpv I agreed -- might be best to not necessarily assume we are using a VM for the sandbox?